src/cookies/__tests__/getCurrentPortalUrlInfo.test.ts

```diff
@@ -0,0 +1,35 @@
+import { getCurrentConnectionInfo } from '../../getCurrentConnectionInfo';
+
+const windowFixture = {
+  location: {
+    hash: '#/',
+    host: 'localhost:8080',
+    hostname: 'localhost',
+    href: 'http://localhost:8080/u/0/explorer/#/',
+    origin: 'http://localhost:8080',
+    pathname: '/u/0/explorer/',
+    port: '8080',
+    protocol: 'http:',
+  },
+} as any as Window;
+
+describe('getCurrentConnectionInfo', () => {
+  it('returns connectionIndex and product of current valid url', () => {
+    const window = { ...windowFixture };
+
+    const info = getCurrentConnectionInfo(window);
+    expect(info).toBeTruthy();
+    expect(info?.connectionIndex).toBe(0);
+    expect(info?.product).toBe('explorer');
+  });
+
+  it('returns null for invalid urls', () => {
+    const window = { ...windowFixture };
+
+    expect(getCurrentConnectionInfo(window)).toBeTruthy();
+
+    window.location.href = 'http://localhost:8080/u/0/';
```

Comment:
Seems like this should be a valid url (for portal itself). So this should match correctly. However a route like `http://foo/notu/0/` should not match even though there is a `u/number/`

Reply:
so you're saying:
* add a ^ starts with requirement like `^/u/...`
* and make the `/u/0/<product>/` product part optional?

Reply:
something like this?
[`dd58d7c` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/dd58d7cdb3c1e4af9c4975267879eaf63c9b6a19)

Reply:
alright, here's the regex update:
[`18ca9ae` (#8)](https://github.com/stardog-union/portal-sdk/pull/8/commits/18ca9ae7461db2c34a180ed48caecc7ce51c9d75)